src/metrics/representation.py

```diff
@@ -146,228 +153,243 @@ def compute(self) -> float:
         Finally, it compute metric.
 
         Returns:
-            Metric value.
+            metric: Metric value.
         """
         vectors = torch.cat(self.vectors).numpy()
-        
         if self.normalize_vectors:
             vectors = normalize(vectors)
 
         if self.dataset_type == DatasetType.CLASSIFICATION:
             # if classification dataset
             targets = torch.cat(self.targets).numpy()
             # prepare data
-            q_vecs, db_vecs, relevants, scores, \
-                db_idxs, q_order_idxs = self.prepare_classification_data(vectors, targets)
+            relevant_idxs, gallery_idxs, query_row_idxs = self.prepare_classification_data(targets)
+            # mock scores and query column indexes because it belong to representation data
+            scores = None
+            query_column_idxs = None
+            # all query vectors is in relevant, so create array with True elements
```

Comment:
I've seen a row `relevant = group.drop(query_idx).tolist()` in `prepare_classification_data`. Doesn't it mean that you removed the queries from relevant already?

Reply:
In classification data i remove query from relevants, not in gallery https://github.com/eora-ai/torchok/pull/53/files#:~:text=ts%20%3D%20pd.Series,relevant_idxs.append(relevant). 
In  representation dataset i not remove query from relevant because the score matrix should contain zeros in diagonal elements (but it's not checked). If i need to check it? 
If you ask why i remove from relevant, i told you because many metric after compute divide metric to `len(relevants)`